mlir/lib/Catalyst/Transforms/TBAAPatterns.cpp

```diff
@@ -31,7 +31,13 @@ void setTag(mlir::Type baseType, catalyst::TBAATree *tree, mlir::MLIRContext *ct
 {
     mlir::LLVM::TBAATagAttr tag;
     if (isa<IndexType>(baseType) || isa<IntegerType>(baseType)) {
-        tag = tree->getTag("int");
+        // Index can be used as a pointer.
+        if (isa<IndexType>(baseType) && (isa<LLVM::StoreOp>(newOp) || isa<LLVM::LoadOp>(newOp))) {
```

Comment:
Here newop should be use just to set the tag, I would remove that `(isa<LLVM::StoreOp>(newOp) || isa<LLVM::LoadOp>(newOp))`